.circleci/config.yml

```diff
@@ -1,109 +1,303 @@
-          key: v1-npmcache-{{ checksum ".nvmrc" }}-{{ checksum "npm-shrinkwrap.json" }}
-          paths:
-            - ~/.npm
+  defaults: &defaults
+    working_directory: ~/wp-calypso
+    docker: *docker-image
+    environment: *shared-environment
 
+version: 2
+jobs:
+  lint-and-translate:
+    <<: *defaults
+    parallelism: 1
+    steps:
+      ###############
+      # Basic Setup #
+      # #############
+      # repo
+      - restore_cache: *restore-git-cache
+      - checkout
+      - run: *update-git-master
+      - save_cache: *save-git-cache
+      # npm dependencies
+      - restore_cache: *restore-node-modules-cache
+      - run: *npm-install
+      - save_cache: *save-node-modules-cache
+      # folders to collect results
+      - run: *setup-results-and-artifacts
+      ###################
+      # End Basic Setup #
+      ###################
       - run:
-          name: Build calypso-strings.pot
-          command: |
-            if [ "$CIRCLE_NODE_INDEX" == "0" ]; then
-              npm run translate; mkdir -p $CIRCLE_ARTIFACTS/translate
-              mv calypso-strings.pot $CIRCLE_ARTIFACTS/translate
-            fi
-
+          name: Lint Config Keys
+          when: always
+          command: npm run lint:config-defaults
       - run:
-          name: Build New Strings .pot
+          name: Lint Client and Server
+          when: always
           command: |
-            if [ "$CIRCLE_NODE_INDEX" == "0" ]; then
-              git clone https://github.com/Automattic/gp-localci-client.git
-              bash gp-localci-client/generate-new-strings-pot.sh $CIRCLE_BRANCH $CIRCLE_SHA1 $CIRCLE_ARTIFACTS/translate
-              rm -rf gp-localci-client
+            # We'll use failure code to do some additional processing and report failure manually
+            set +o errexit
+            ./node_modules/.bin/eslint-eslines                        \
+              $(                                                      \
+                git diff --name-only --diff-filter=d origin/master... \
+                | grep -E '^(client|server)'                          \
+                | grep -E '\.jsx?$'                                   \
+              )                                                       \
+              -- --format junit                                       \
+              > ~/eslint-results.xml
+            # Ensure we don't upload empty reports.
+            # Eslint would not produce a malformed empty xml using the --format and -o options.
+            # However, with eslint-eslines and the redirected > output, we may get an empty file.
+            # We only put the file into the reports directory if the status code informs us that
+            # lint issues have been encountered.
+            ESLINT_RESULT="$?"
+            if [ "$ESLINT_RESULT" -ne 0 ]; then
+              mv ~/eslint-results.xml "$CIRCLE_TEST_REPORTS/eslint/results.xml"
+              exit "$ESLINT_RESULT"
```

Comment:
Do we need to delete `~/eslint-results.xml` after this?

Reply:
If there is a report, we move the file to the test_results so it will be uploaded and we need to keep it.

If there isn't a report, it shouldn't matter. These containers are ephemeral and `eslint-results.xml` will wind in the ether where all deleted files go.